lms/djangoapps/instructor/views/api.py

```diff
@@ -519,8 +519,7 @@ def generate_random_string(length):
         char for char in string.ascii_uppercase + string.digits + string.ascii_lowercase
         if char not in 'aAeEiIoOuU1l'
     ]
-
-    return string.join((random.choice(chars) for __ in range(length)), '')
+    return ''.join((random.choice(chars) for i in range(length)))
```

Comment:
string.join to ''.join and replace __ with i to prevent recursive infinite loop in py3.